scripts/vet.sh

```diff
@@ -148,6 +148,7 @@ Target is deprecated: Use the Target field in the BuildOptions instead.
 UpdateAddresses is deprecated:
 UpdateSubConnState is deprecated:
 balancer.ErrTransientFailure is deprecated:
+internal.Logger is deprecated:
```

Comment:
Done. But I also had to leave the current line in there.
- New check with fail if there are any usages of the deprecated `Logger` interface from anywhere other than `prefix_logger.go`.
- Once that check passes, this will ensure that the usage in `prefix_logger.go` does not cause vet to fail.